### PR TITLE
feat: add Windows CI back

### LINKED ISSUE #1274: [FEATURE] Add Windows builds back
### Detailed description

Many people still develop and test algorithms on Windows, not only in Linux or MacOS.
However, this repository only has Linux and MacOS support, which if not well-tested on Windows, the code may have errors.

I suggest adding back the Windows builds. There are a few errors, but those can be fixed.

### Context

1. Windows users do not have to worry about errors or unwanted things happening when building.
2. This ensures the code works perfectly fine on the latest Windows versions.
3. MSVC (sometimes) catches a few other errors the other compilers do not.

### Possible implementation

Add Windows builds back.

### Additional information

I'd like to work on this, if agreed to add them back.

### PR DESCRIPTION
**Pull request by Panquesito7**
#### Description of Change

- Add Windows builds back.
  - Thanks to @realstealthninja for their help!
  - Closes #1274.

#### Checklist

- [x] Added description of change
- [ ] ~Added file name matches [File name guidelines](https://github.com/TheAlgorithms/C/blob/master/CONTRIBUTING.md#File-Name-guidelines)~
- [ ] ~Added tests and example, test must pass~
- [ ] ~Relevant documentation/comments is changed or added~
- [x] PR title follows semantic [commit guidelines](https://github.com/TheAlgorithms/C/blob/master/CONTRIBUTING.md#Commit-Guidelines)
- [x] Search previous suggestions before making a new one, as yours may be a duplicate.
- [x] I acknowledge that all my contributions will be made under the project's license.

Notes:

### CODE REVIEW

File: searching/exponential_search.c [hunk 1]
```diff
@@ -81,7 +82,7 @@ int main()
 static void test()
 {
     // empty array
-    int64_t arr_empty[] = {};
+    int64_t arr_empty[] = { 0 };
     assert(exponential_search(arr_empty, 0, 10) == -1);
```
Comment (realstealthninja) [inline searching/exponential_search.c:86]: This might fail

**Review by realstealthninja — CHANGES_REQUESTED** [review]

File: searching/exponential_search.c (hunk 1 above)
Comment (Panquesito7) [reply searching/exponential_search.c:86]: It didn't. I tested it on Ubuntu/Gitpod and it worked fine.

Comment (realstealthninja) [reply searching/exponential_search.c:86]: i see alright!

**Review by realstealthninja — APPROVED** [review]

**Comment by Panquesito7** [comment]
Will be merging in a few days if there are no objections. Ping me in case I forget.

**Review by alexpantyukhin — APPROVED** [review]
Cool!

**Pull request merged**